### PR TITLE
[PM-11137] Implement iOS 18 Totp autofill from list

### PR DESCRIPTION
**Pull request by fedemkr**
## 🎟️ Tracking

[PM-11137](https://bitwarden.atlassian.net/browse/PM-11137)

## 📔 Objective

Implement autofill Totp from list -> One time codes from [Streamline sign-in with passkey upgrades and credential managers](https://developer.apple.com/videos/play/wwdc2024/10125/).

### Notes

⚠️  This is an iOS 18 feature. So this requires Xcode 16 to be compiled.

## 📸 Screenshots

https://github.com/user-attachments/assets/3aed8f7d-af06-4081-8af0-b79b4ef2ef9d

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11137]: https://bitwarden.atlassian.net/browse/PM-11137?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by speedy1804** [comment]
hey I'm not that familiar with Github: Have you coded this now so that it could be implemented?

**Comment by fedemkr** [comment]
⛏️  Fixing tests...

**Review by matt-livefront — COMMENTED** [review]
Does this also need to support autofilling the code from the keyboard suggestions or is that being done separately?

File: BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListProcessor.swift [hunk 1]
```diff
@@ -209,6 +254,41 @@ class VaultAutofillListProcessor: StateProcessor<
...
+            let refreshedSearchResults = try await services.vaultRepository.refreshTOTPCodes(for: searchItems)
+            let allSearchResults = currentSearchResults.updated(with: refreshedSearchResults)
+            searchTotpExpirationManager?.configureTOTPRefreshScheduling(for: allSearchResults)
+            state.ciphersForSearch = [
+                VaultListSection(id: "", items: allSearchResults, name: ""),
+            ]
+        } catch {
+            services.errorReporter.log(error: error)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListProcessor.swift:290]: 👍 Yep, makes total sense. I'll extract this to a common place and apply it as well in the `VaultGroupProcessor`.

File: BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListView.swift [hunk 2]
```diff
@@ -202,17 +209,21 @@ private struct VaultAutofillListSearchableView: View {
...
-                            Label {
-                                Text(store.state.emptyViewButtonText)
-                            } icon: {
-                                Asset.Images.plus16.swiftUIImage
-                                    .imageStyle(.accessoryIcon(
-                                        color: Asset.Colors.buttonFilledForeground.swiftUIColor,
-                                        scaleWithFont: true
-                                    ))
+                        if store.state.isAutofillingFido2List {
+                            EmptyView()
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListView.swift:213]: Nice catch I meant to use `isAutofillingTotpList`

**Comment by fedemkr** [comment]
@matt-livefront  The implementation from the QuickType bar will be done in a separate PR of this ticket [PM-14497](https://bitwarden.atlassian.net/browse/PM-14497)

File: BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListState.swift [hunk 3]
```diff
@@ -43,4 +46,9 @@ struct VaultAutofillListState: Equatable, Sendable {
 
     /// The list of sections to display for matching vault items.
     var vaultListSections = [VaultListSection]()
+
+    /// Whether to show the add item button.
+    var showAddItemButton: Bool {
+        !isAutofillingFido2List && !isAutofillingTotpList
```
Comment (matt-livefront) [inline BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListState.swift:52]: ❓ Do we want to hide the add item button in the navigation bar when `isAutofillingFido2List`?

Related conversation: https://github.com/bitwarden/ios/pull/884#discussion_r1833162574

Comment (fedemkr) [reply BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListState.swift]: Nice catch, fixed! Sorry I think I did some of the work while working on the Beta and the change of Fido2 showing the button was made afterwards and that's why I have some of that stuff mixed up. Thanks again for catching all of this and sorry for the inconvenience!

Comment (matt-livefront) [reply BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListState.swift]: 👍 No problem!

**Review by matt-livefront — APPROVED** [review]

**Comment by speedy1804** [comment]
I just saw it has been implemented - great work! when will this be available in the beta app?

**Comment by Gerardv514** [comment]
Idk if this pr changes is related, but I recently had to open a git issue bug. My bw app is no longer auto copying Totp codes after an iOS password autofill when the vault timeout is set to never. 

Noticed this issue for about a month or two now, never had the problem before. 

https://github.com/bitwarden/ios/issues/1634

**Pull request merged**